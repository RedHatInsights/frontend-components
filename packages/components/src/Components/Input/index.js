export { default as Input } from './Input';
export { default as LabeledInput } from './LabeledInput';
