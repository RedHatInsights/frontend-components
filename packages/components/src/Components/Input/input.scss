@import '~@patternfly/patternfly/components/FormControls/form-control.css';
@import '~@patternfly/patternfly/components/Check/check.scss';
