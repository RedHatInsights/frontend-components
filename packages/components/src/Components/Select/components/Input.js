import React from 'react';
import { components } from 'react-select';
import PropTypes from 'prop-types';

const Input = (props) => (
    <components.Input {...props}  />
);

Input.propTypes = {
    selectProps: PropTypes.shape({
        isMulti: PropTypes.bool,
    }).isRequired,
};

export default Input;
