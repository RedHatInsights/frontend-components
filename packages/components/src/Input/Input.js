import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';

const checkTypes = [ 'checkbox', 'radio' ];

const Input = ({ type = 'text', ariaLabel = type, className, ...props }) => {
    const classes = checkTypes.indexOf(type) !== -1 ? 'pf-c-check' : 'pf-c-form-control';
    return (
        <input { ...props }
            widget-type='InsightsInput'
            type={ type }
            aria-label={ ariaLabel }
            className={ classnames(classes, className) }
        />
    );
};

Input.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    ariaLabel: PropTypes.string
};

Input.defaultProps = {
    'widget-id': 'ins-component-table'
};

export default Input;
