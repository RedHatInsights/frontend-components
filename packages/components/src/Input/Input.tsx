import React from 'react';
import classnames from 'classnames';
import { useOUIAId } from '@patternfly/react-core';

export interface InputProps extends Omit<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, 'capture'> {
  type?: string;
  className?: string;
  ariaLabel?: string;
  ouiaId?: string;
  ouiaSafe?: boolean;
  'data-ouia-component-type'?: string;
  'data-ouia-component-id'?: string;
  'data-ouia-safe'?: boolean;
  'widget-id'?: string;
}

const checkTypes = ['checkbox', 'radio'];

const Input: React.FC<InputProps> = ({ type = 'text', ariaLabel = type, className, ouiaId, ouiaSafe = true, ...props }) => {
  const classes = checkTypes.indexOf(type) !== -1 ? 'pf-c-check' : 'pf-c-form-control';
  const ouiaComponentType = 'RHI/Input';
  const ouiaFinalId = useOUIAId(ouiaComponentType, ouiaId, ouiaSafe as unknown as string);
  return (
    <input
      {...props}
      data-ouia-component-type={ouiaComponentType}
      data-ouia-component-id={ouiaFinalId}
      data-ouia-safe={ouiaSafe}
      type={type}
      aria-label={ariaLabel}
      className={classnames(classes, className)}
    />
  );
};

export default Input;
