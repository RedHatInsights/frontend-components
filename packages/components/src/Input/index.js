export { default } from './Input';
export { default as Input } from './Input';
export { default as LabeledInput } from './LabeledInput';
