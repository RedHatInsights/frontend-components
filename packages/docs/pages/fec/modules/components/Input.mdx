
# Input

## Props

|name|type|default|description|
|----|----|-------|-----------|
|type|`string`|`'text'`||
|className|`string`|||
|ariaLabel|`string`|`type`||
|ouiaId|`string`|||
|ouiaSafe|<code>&#123;<br/><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span> "name": "boolean"<br />&#125;</code>|`true`||
|data-ouia-component-type|`string`|||
|data-ouia-component-id|`string`|||
|data-ouia-safe|<code>&#123;<br/><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span> "name": "boolean"<br />&#125;</code>|||
|widget-id|`string`|||



