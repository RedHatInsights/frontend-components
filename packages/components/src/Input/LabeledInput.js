import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import Input from './Input';

const LabeledInput = ({ type = 'text', className, children, name, ...props }) => {
    // Might generate ID which is not unique enough
    name = name || new Date().getTime() + Math.random().toString(36);
    return (
        <label className={ classnames('pf-c-form__label', className) } htmlFor={ name }>
            <Input { ...props } type={ type } name={ name } id={ name } /> { children }
        </label>
    );
};

LabeledInput.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.node,
    name: PropTypes.string
};

export default LabeledInput;
