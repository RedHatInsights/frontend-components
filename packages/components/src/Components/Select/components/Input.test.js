import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from './Input';

describe('Input component', () => {
    describe('should render', () => {
        [true, false].map(item => {
            it(`isMulti - ${item}`, () => {
            const wrapper = shallow(<Input selectProps={{ isMulti: item }} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        })});
    });
});
