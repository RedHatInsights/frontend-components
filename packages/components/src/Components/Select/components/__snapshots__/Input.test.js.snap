// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render isMulti - false 1`] = `
<Input
  selectProps={
    Object {
      "isMulti": false,
    }
  }
/>
`;

exports[`Input component should render isMulti - true 1`] = `
<Input
  selectProps={
    Object {
      "isMulti": true,
    }
  }
/>
`;
